transit_odp/timetables/extract.py

```diff
@@ -316,6 +318,103 @@ def extract_journey_patterns(self):
 
         return journey_patterns, jp_to_jps
 
+    def extract_journey_rout_link(self):
+        # Get services and journey patterns
+        services = self.doc.get_services()
+        journey_patterns = journey_patterns_to_dataframe(services, False)
+
+        # Create a DataFrame with pattern_id and route_ref
+        jp_route_ref = journey_patterns[["pattern_id", "route_ref"]]
+
+        # Group by route_ref and create a list of pattern_ids
+        df_vj_grouped = (
+            jp_route_ref.groupby("route_ref")["pattern_id"].apply(list).reset_index()
+        )
+        df_vj_grouped = df_vj_grouped.rename(columns={"pattern_id": "jp_ref"})
+
+        # Get routes and create a route reference link dictionary
+        routes = self.doc.get_route()
+        route_ref_link = {}
+
+        for route in routes:
+            route.text
+            route_section_refs = route.get_elements_or_none(["RouteSectionRef"])
+            route_ref_ids = []
+
+            # Extract route section references and update route_ref_link dictionary
+            for route_section_ref in route_section_refs:
+                route_id = route.get("id")
+                if route_section_ref and route_id:
+                    route_section_ref_text = (
+                        route_section_ref.text if route_section_ref else ""
+                    )
+                    route_ref_ids.append(route_section_ref_text)
+
+            route_ref_link.update({route.get("id"): route_ref_ids})
+
+        # Convert route_ref_link dictionary to DataFrame
+        route_map = pd.DataFrame(
```

Comment:
let us move this section to new function for better understanding of the functin